tuf_conformance/repository_simulator.py

```diff
@@ -274,41 +290,29 @@ def fetch_metadata(self, role: str, version: int | None = None) -> bytes:
 
         If version is None, non-versioned metadata is being requested.
         """
+
         # decode role for the metadata
         role = parse.unquote(role, encoding="utf-8")
 
         if role == Root.type:
-            # return a version previously serialized in publish_root()
-            if version is None or version > len(self.signed_roots):
+            if version is None or version > len(self.signed_mds[Root.type]):
                 raise ValueError(f"Unknown root version {version}")
-            logger.debug("fetched root version %d", version)
-            return self.signed_roots[version - 1]
-
-        # sign and serialize the requested metadata
-        md = self.mds.get(role)
-
-        if md is None:
-            raise ValueError(f"Unknown role {role}")
-
-        md.signatures.clear()
-        for signer in self.signers[role].values():
-            md.sign(signer, append=True)
-
-        logger.debug(
-            "fetched %s v%d with %d sigs",
-            role,
-            md.signed.version,
-            len(self.signers[role]),
-        )
-        return md.to_bytes(JSONSerializer())
+            return self.signed_mds[Root.type][version - 1]
+        # Non-root mds:
+        if len(self.signed_mds[role]) == 0:
+            raise ValueError(f"The repository has not published metadata for '{role}'")
+        if version is not None:
+            return self.signed_mds[role][version - 1]
+        return self.signed_mds[role][-1]
 
     def _version(self, role: str) -> int:
         signed = self.mds[role].signed
         assert isinstance(signed, Root | Timestamp | Snapshot | Targets)
         return signed.version
 
     def _compute_hashes_and_length(self, role: str) -> tuple[dict[str, str], int]:
-        data = self.fetch_metadata(role)
+        md = Metadata.from_bytes(self.signed_mds[role][-1])
+        data = md.to_bytes(JSONSerializer())
```

Comment:
deserialization is not needed  and is in fact actively harmful (since we want to be able to put invalid data into `signed_mds[role]`).

```suggestion
        data = self.signed_mds[role][-1]
```

Reply:
https://github.com/theupdateframework/tuf-conformance/pull/188/commits/fd4d96b7394a4fff6e68d9beed2e7d81ae973bf1